padatious/intent_container.py

```diff
@@ -116,10 +135,13 @@ def train(self, *args, **kwargs):
                 each time a new intent is trained
             single_thread (bool): Whether to force running in a single thread
         """
-        self.intents.train(*args, **kwargs)
-        self.entities.train(*args, **kwargs)
-        self.entities.calc_ent_dict()
         self.padaos.compile()
+
+        timeout = kwargs.setdefault('timeout', 20)
+        self.train_thread = Thread(target=self._train, args=args, kwargs=kwargs, daemon=True)
+        self.train_thread.start()
+        self.train_thread.join(timeout)
+
```

Comment:
What would happen if training stalls like was reported by @penrods and training is called again? (In Mycroft I think this happens if a skill is reloaded)

Reply:
Nice point to bright up! I've added a test case for that and made it simply skip training.

Reply:
Does it skip the second request?  Or kill the stalled training session?

Reply:
After the timeout, it tries to kill the training, but due to Python's weirdness, it doesn't seem possible to kill it. So any subsequent calls to train will simply be ignored (and it will instead only return exact matches).

Reply:
Then we need a better solution, as this has certainly happened several times and it is a real user problem when it happens.

I know it is draconian, but I'd rather force a reboot than leave the user in a half-broken state.

Reply:
Simplest may be to just raise a TimeoutError and let the padatious service in mycroft send a reboot message or if reboot is not possible ask the user to restart...

Reply:
I've pushed a change that returns the status of whether it properly trained so we could do this, but an alternative solution is that we add a simple `padatious-train` command line script that writes the intents to the cache. Since this would be called as subprocess, the daemon threads would get killed when the main thread returns.